File: swift/Sources/K256/Types/OrderLevel.swift
import Foundation

/// Order book level with price and size.
public struct K256OrderLevel: Sendable, Codable, Equatable {
    /// Price in base units
    public let price: UInt64
    /// Size in base units
    public let size: UInt64

    public init(price: UInt64, size: UInt64) {
        self.price = price
        self.size = size
    }
}

File: php/src/Types/OrderLevel.php
<?php

declare(strict_types=1);

namespace K256\Types;

/**
 * Order book level with price and size.
 */
readonly class OrderLevel
{
    public function __construct(
        /** Price in base units */
        public int $price,
        /** Size in base units */
        public int $size,
    ) {}
}
